src/redis_mod.f90

```diff
@@ -60,22 +71,83 @@ subroutine stream_data(f)
 
   end subroutine stream_data
 
-  subroutine redis_push(c, key, arr, dtype, dims)
```

Comment:
I think we should probably leave this generic interface in place . Maybe we can change the name to `redis_push_explicit` or something, and then call it in the various `redis_push_f4` subroutines.